os/drivers/lcd/mipi_lcd.c

```diff
@@ -129,13 +129,25 @@ struct mipi_lcd_dev_s {
 
 static struct mipi_lcd_dev_s g_lcdcdev;
 
-static int send_cmd_retry(struct mipi_lcd_dev_s *priv, struct mipi_dsi_msg* msg)
+static int send_cmd_transfer(struct mipi_lcd_dev_s *priv, struct mipi_dsi_msg* msg)
 {
 	int transfer_status = ERROR;
 	int max_cmd_retry_count = CONFIG_LCD_SEND_CMD_RETRY_COUNT;
-	while (max_cmd_retry_count && transfer_status != OK) {
-		transfer_status = mipi_dsi_transfer(priv->dsi_dev, msg);
-		max_cmd_retry_count--;
+	transfer_status = mipi_dsi_transfer(priv->dsi_dev, msg);
+
+	if (transfer_status != OK) {
+		/* Retry command send */
+		if (CONFIG_LCD_SEND_CMD_RETRY_COUNT > 0) {
+			while (max_cmd_retry_count && transfer_status != OK) {
+				transfer_status = mipi_dsi_transfer(priv->dsi_dev, msg);
+				max_cmd_retry_count--;
+			}
+			if (transfer_status != OK) {
+				lcddbg("Command %x not sent after retry\n", cmd);
+			}
+		} else {
+			lcddbg("Command %x not sent\n", cmd);
+		}
```

Comment:
I think `if (CONFIG_...._COUNT > 0)` is unnecessary. The while conditional covers it.
Could you check it?